Szwalnia/DodajDostawce.cs

```diff
@@ -48,10 +48,31 @@ private void btnDodaj_Click(object sender, EventArgs e)
                     dostawcaNowy.Telefon_2 = Convert.ToInt32(mtxtTel2.Text);
                 }
                 dostawcaNowy.Email = txtEmail.Text;
-                
-                db.Dostawcy_Zaopatrzenie.Add(dostawcaNowy);
-                db.SaveChanges();
-                MessageBox.Show("Dodano nowego dostawcę.");
+
+                if(!db.Dostawcy_Zaopatrzenie.Where(dostawca => dostawca.Nazwa == txtNazwa.Text).Any())
```

Comment:
proszę spróbować taki kod:

`if(!this.isSupplierValid(){
	return;
}				
db.Dostawcy_Zaopatrzenie.Add(dostawcaNowy);
db.SaveChanges();
MessageBox.Show("Dodano nowego dostawcę.");

private bool isSupplierValid(){
	if(db.Dostawcy_Zaopatrzenie.Where(dostawca => dostawca.Nazwa == txtNazwa.Text).Any()){
	MessageBox.Show("Dostawca o takiej nazwie już istnieje");
	return false;
}

 if (db.Dostawcy_Zaopatrzenie.Where(dostawca => dostawca.Telefon_1 == dostawcaNowy.Telefon_1).Any()) {
	 MessageBox.Show("Dostawca z takim numerem telefonu już istnieje.");
	return false;
 }

 if (db.Dostawcy_Zaopatrzenie.Where(dostawca => dostawca.Email == txtEmail.Text).Any()){
	 MessageBox.Show("Dostawca z takim adresem e-mail już istnieje.");
	return false;
 }
      return true;  
}`